src/transformer.js

```diff
@@ -54,74 +89,111 @@ const transform = (file, api, options) => {
-      const info = getAbsPathInfoSync(absPath);
-      let replacement;
-      if (info === null) {
-        // No info was found, so "don't replace it".
-        replacement = importPath;
-      } else if (info.isFile) {
-        // Join the import path directory with the real filename from the info.
-        replacement = path.join(path.dirname(importPath), path.basename(info.path));
-      } else {
-        // If it's a directory, call the function that checks for a `package.json` or an `index`.
-        const folderReplacement = createReplacementForFolder(absPath, importPath);
-        replacement = folderReplacement || importPath;
-      }
-
-      /**
-       * This is a hotfix; when you use `path.join()` with a path that starts with `./`, the
-       * function removes it, but that's needed for import statements: if they path doesn't
-       * starts with `.`, it assumes that it's on `node_modules`.
-       */
-      if (
-        (importPath === '.' || importPath.startsWith('./')) &&
-        !replacement.startsWith('./')
-      ) {
-        replacement = `./${replacement}`;
-      }
+      const replacement = updatePath(item, base);
 
       // Replace the node with a new one on the AST.
+      // return j.importDeclaration.from({ ...item.value, source: j.literal(replacement) });
       return j.importDeclaration(item.value.specifiers, j.literal(replacement));
     });
+  root
+    .find(j.ExportNamedDeclaration)
+    // Filter out the ones that are on the ignore list.
+    .filter(needsPathUpdate)
+    .replaceWith((item) => {
+      const replacement = updatePath(item, base);
+
+      // Replace the node with a new one on the AST.
+      return j.exportNamedDeclaration.from({
+        ...item.value,
+        source: j.literal(replacement),
+      });
+      // (
+      //   item.value.declaration,
+      //   item.value.specifiers,
+      //   j.literal(replacement),
+      // );
+    });
+  root
+    .find(j.ExportAllDeclaration)
+    // Filter out the ones that are on the ignore list.
+    .filter(needsPathUpdate)
+    .replaceWith((item) => {
+      const replacement = updatePath(item, base);
+
+      // Replace the node with a new one on the AST.
+      return j.exportAllDeclaration.from({
+        ...item.value,
+        source: j.literal(replacement),
+      });
+    });
   // =================================================
   // Parse the modules modifications.
   // =================================================
   if (cjs2esm.modules.length) {
+    const modules = cjs2esm.modules.map((r) => ({
+      ...r,
+      regex: new RegExp(`^${r.name}(/|$)`),
+    }));
+
     root
       .find(j.ImportDeclaration)
       // Filter out the import statments that don't need to be modified.
       .filter((item) => {
         const importPath = item.value.source.value;
-        return cjs2esm.modules.some((mod) => importPath.startsWith(mod.name));
+        return modules.some((mod) => mod.regex.test(importPath));
```

Comment:
```suggestion
        return modules.some((mod) => importPath.match(mod.regex));
```

Yes, I'm being picky here: even though you didn't use `g`, I only use `.test` when I need the `lastIndex`.